ghermaico135/react-capstone-project
# React capstone project - Metrics webapp

- This React capstone project  is about building a mobile web application to check a list of metrics (numeric values) that  making use of React and Redux.
- The webapp will have several pages :page_facing_up: 
- Homepage.:one page with a list of items that could be filtered by some parameters
- Detailspage-Detail list of the country item

- [ ] General requirements:

- Make sure that there are [no linter errors]
- Make sure that you used correct [Gitflow]

- [ ] Project requirements 

- choose an API to base the development of the webapp on. 
- The API should allow you to get numeric values based on a parameter.

- [ ] Interactions :page_facing_up: 

1. Home page :page_with_curl: 

- When the page loads, the webapp shows the list of categories that could be filtered out by some parameter.
- Along with the category name, you will display some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.

2. Details page :page_facing_up: 

- In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button , the user navigates to the home page.

- [ ] Testing requirements 

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API, so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.

- [ ] Technical requirements 

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online 
- Add a link to the online version of your app to your README file.

- [ ] Project documentation

Record a video presenting the features of the project 

- a description of the project.
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.
